actors/builtin/miner/miner_actor.go

```diff
@@ -319,20 +317,13 @@ func (a Actor) ProveCommitSector(rt Runtime, params *ProveCommitSectorParams) *a
 		rt.Abortf(exitcode.ErrNotFound, "no precommitted sector %v", sectorNo)
 	}
 
-	if rt.CurrEpoch() > precommit.PreCommitEpoch+PoRepMaxDelay || rt.CurrEpoch() < precommit.PreCommitEpoch+PoRepMinDelay {
-		rt.Abortf(exitcode.ErrIllegalArgument, "Invalid ProveCommitSector epoch")
-	}
-
-	TODO()
-	// TODO HS: How are SealEpoch, InteractiveEpoch determined (and intended to be used)?
-	// Presumably they cannot be derived from the SectorProveCommitInfo provided by an untrusted party.
-
 	a.verifySeal(rt, st.Info.SectorSize, &abi.OnChainSealVerifyInfo{
-		SealedCID:    precommit.Info.SealedCID,
-		SealEpoch:    precommit.Info.SealEpoch,
-		Proof:        params.Proof,
-		DealIDs:      precommit.Info.DealIDs,
-		SectorNumber: precommit.Info.SectorNumber,
+		SealedCID:        precommit.Info.SealedCID,
+		InteractiveEpoch: precommit.PreCommitEpoch + PreCommitChallengeDelay,
```

Comment:
This is a stricter check than before, as this has now to be drawn from a specific block, but it seems like a good change from my side.